### PR TITLE
Allow user to define ACLs for OpenStack keys

### PR DESCRIPTION
**Pull request by fultonj**
The openstack_keys structure now supports a key called acls whose value is a list of strings that one could 
pass to setfacl. The ansible ACL module applies the ACLs to all openstack keys with this property.

Fixes: #1688

### CODE REVIEW

**Comment by fultonj** [comment]
Of the 18 failing tests:

- 13 were "vagrant: command not found"
- 1 was "org.kohsuke.github.HttpException: Server returned HTTP response code: -1,"
  (ceph-ansible-prs-luminous-ansible2.2-docker_dmcrypt_journal_collocation)

For the four that actually ran the job I will send a new PR. Two of them uncovered a case that should have been obvious to me before I sent the PR (when acls isn't set): 

- https://jenkins.ceph.com/job/ceph-ansible-prs-luminous-ansible2.2-docker_cluster/186/console
- https://jenkins.ceph.com/job/ceph-ansible-prs-luminous-ansible2.2-purge_docker_cluster/185/console

The other two are permissions errors and likely related to key creation and legit bugs for my PR: `librados: client.bootstrap-osd authentication error (1) Operation not permitted`

- https://jenkins.ceph.com/job/ceph-ansible-prs-luminous-ansible2.2-bluestore_cluster/182/console
- https://jenkins.ceph.com/job/ceph-ansible-prs-jewel-ansible2.2-update_cluster/778/console

**Comment by guits** [comment]
jenkins test this please

File: roles/ceph-client/tasks/create_users_keys.yml [hunk 1]
```diff
@@ -30,3 +30,17 @@
...
+
+- name: setfacl for key(s)
+  acl:
+    path: "/etc/ceph/{{ cluster }}.{{ item.0.name }}.keyring"
+    entry: "{{ item.1 }}"
+    state: present
+  with_subelements:
+    - "{{ keys }}"
+    - acls
+    - flags:
```
Comment (guits) [inline roles/ceph-client/tasks/create_users_keys.yml]: can simply be written this way :
```
with_subelements:
  - "{{ keys }}"
  - acls
  - skip_missing: true
```

File: roles/ceph-mon/tasks/openstack_config.yml [hunk 2]
```diff
@@ -49,3 +49,17 @@
...
+
+- name: setfacl for openstack key(s)
+  acl:
+    path: "/etc/ceph/{{ cluster }}.{{ item.0.name }}.keyring"
+    entry: "{{ item.1 }}"
+    state: present
+  with_subelements:
+    - "{{ openstack_keys }}"
+    - acls
+    - flags:
```
Comment (guits) [inline roles/ceph-mon/tasks/openstack_config.yml]: can simply be written this way :
```
with_subelements:
  - "{{ keys }}"
  - acls
  - skip_missing: true
```

**Comment by guits** [comment]
jenkins test jewel-ansible2.2-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-bluestore_docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-xenial_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.2-xenial_cluster

**Pull request merged**